### ISSUE TITLE
Feature Request : Filter functionality in our beacons

### ISSUE THREAD

**Issue by Arjun2905** [CCExtractor/beacon#155]
**Is your feature request related to a problem? Please describe.**
Suppose in crew of several members different people are able to create a beacon. It will become a tedious task to look upon every beacon to search a particular beacon.

**Describe the solution you'd like**
Here the filter functionality can play a vital role to optimize searching based on the person who had created a beacon. Currently in every beacon card has beacon name and who had made it as **"Beacon Name by Creator Name"**  So at the time when we take information from backend about every beacon and display it on cards we will store the owner names in a set and use these names to provide filter functionality.
 
**Describe alternatives you've considered**
Alternative to this can be to implement search bar where user can search a particular beacon name. But in this case we need to know perfect name of beacon otherwise we'll end up no result found.

**Additional context**
1. Use of Icon Button (with provided button color in theme as yellow).
2. Use of pop-up menu by onTap on icon button.
3. Showing the list of all beacon owners of whom we have joined the beacons from earlier(like as if we have registered it).

**Are you working on this? (Yes/No)**
Yes

**Comment by Arjun2905** [comment]
@nb9960 would like to hear some of your suggestion's on it.

**Comment by cfsmp3** [comment]
This is a useful feature for sure; however searching by "name" is going to be incomplete, if name here means the name of the person.

Suppose this very common use case: A hike group that goes out every weekend have several leaders, and one of them takes the group out each day. There's many people that can go on these hikes (as "followers") who may or may not go.

My point here is that we'd need to introduce the concept of "group" (which I believe we still don't have) and followers should be able to follow groups or search for group; you don't really care who created the beacon but rather who has the beacon for a group.

It's definitely a good idea; just need some refining so it can adjust to the actual use cases.

**Comment by Arjun2905** [comment]
@cfsmp3 yeah according to the existing codebase your point seems to be completely valid over here, Earlier I was thinking to displaying names of those names people only who have created a beacon but thinking on your advice I guess it will just clatter the names of all people who have created any beacon and for that we need groups functionality for that. I'll definitely try to put your advice in well planned manner in my proposal.
